localstack-core/localstack/config.py

```diff
@@ -429,6 +429,10 @@ def in_docker():
 LS_LOG = eval_log_type("LS_LOG")
 DEBUG = is_env_true("DEBUG") or LS_LOG in TRACE_LOG_LEVELS
 
+# PUBLIC: 0 (default)
+# When enabled it triggers specialised workflows for the debugging.
+DEBUG_MODE = is_env_true("DEBUG_MODE")
+
```

Comment:
this may be potentially confusing to people, since we already have `DEBUG=1`. if this is specific to lambda, i would suggest `LAMBDA_DEBUG_MODE`, or if it's supposed to be a generic config, then we should discuss what exactly this config variable does and avoid potential overlap with `DEBUG=`

Reply:
Thank you! As this targets debugging of lambda functions, I agree this is best if renamed.

Reply:
+1
Everyone associates debug mode with `DEBUG=1` (common terminology in support and customer interaction) and we should avoid naming overload.

I'm ok with `LAMBDA_DEBUG_MODE` as preview for now given it's only used in Lambda, but it makes sense to think about the naming here because we envision a LocalStack-wide debug mode that re-configures other services to work smoothly with interactive debugging (e.g., disabling timeouts on upstream services invoking a debug-enabled function).

Brainstorming some alternatives:
* `REMOTE_DEBUGGING` matching the [LS docs](https://docs.localstack.cloud/user-guide/lambda-tools/debugging/)
* `DEBUG_SETTINGS`
* `DEBUG_CONFIG`
* `DEBUGGING_MODE`
* `INTERACTIVE_DEBUG_MODE`
* `INTERACTIVE_DEBUGGING`
* `DEBUG_TIMEOUT_MODE` if it only affects timeouts

What do you think @dfangl @dominikschubert ?